ncclient/operations/third_party/iosxe/rpc.py

```diff
@@ -1,9 +1,500 @@
+        if not self._parsed:
+            self.parse()
+        return self._subscription_result
+
+    @property
+    def subscription_result_xml(self):
+        "*subscription-result* element as an XML string"
+        if not self._parsed:
+            self.parse()
+        return to_xml(self._subscription_result)
+
+    @property
+    def subscription_id(self):
+        "*subscription-id* element as an :class:`~xml.etree.ElementTree.Element`"
+        if not self._parsed:
+            self.parse()
+        if self._subscription_id is not None:
+            return int(self._subscription_id.text)
+        else:
+            return None
+
+    @property
+    def subscription_id_ele(self):
+        "*subscription-id* element as an :class:`~xml.etree.ElementTree.Element`"
+        if not self._parsed:
+            self.parse()
+        return self._subscription_id
+
+    @property
+    def subscription_id_xml(self):
+        "*subscription-id* element as an XML string"
+        if not self._parsed:
+            self.parse()
+        return to_xml(self._subscription_id)
+
+
+class EstablishSubscription(RPC):
+    
+    "`establish-subscription` RPC"
+
+    DEPENDS = ['urn:ietf:params:netconf:capability:notification:1.1']
+    REPLY_CLS = EstablishSubscriptionReply
+    
+    def request(self, callback, errback,
+                xpath=None,
+                period=None, dampening_period=None,
+                streamns=None, streamident=None):
+        """Create a simple subscription for ietf-yang-push subscriptions.
+
+        *callback* user-defined callback for notifications
+
+        *errback* user-defined error handling callback
+
+        *xpath* specifies the xpath-filter element
+
+        *period* period for polling; on-change implied if not set
+
+        *dampening_period* dampening period for change events
+
+        *streamns* XML namespace for a non-default stream identifier
+
+        *streamident* Non-default stream identity
+
+        RPC currently supports only stream `yang-push`.
+        """
+        #
+        # validate parameters
+        #
+        if xpath is None:
+            raise YangPushError("Must have xpath")
+        if period and dampening_period:
+            raise YangPushError("Can only have one of period and dampening_period")
+        if (period and streamident) or (dampening_period and streamident):
+            raise YangPushError("Cannot combine custom stream with periodic or on-change")
+        if (period is None) and (dampening_period is None) and (streamident is None):
+            raise YangPushError("Must have at least one of period, dampening_period or streamident")
+        if streamident and not streamns:
+            raise YangPushError("Must specify namespace for custom stream")
+
+        #
+        # Try to construct request the "standard" way. However,
+        # doesn't work because cannot force lxml to include namespace
+        # values for identities.
```

Comment:
If you want to explore this some more, I think this is where the [`lxml.builder.ElementMaker`](https://lxml.de/api/lxml.builder.ElementMaker-class.html) factory comes in - would be something like:

```python
ele_maker = ElementMaker(namespace=IETF_EVENT_NOTIFICATIONS_NS, nsmap={"yp": IETF_YANG_PUSH_NS})
node = ele_maker("establish-subscription")
stream = sub_ele_ns(node, "stream", IETF_EVENT_NOTIFICATIONS_NS)
...
```